strncpy warnings with GCC 8

WIth GCC 8 I get the following warning on compilation:

```In file included from ../../dpf/distrho/DistrhoPluginMain.cpp:29:
In function 'void DISTRHO::strncpy(char*, const char*, size_t)',
    inlined from 'intptr_t DISTRHO::PluginVst::vst_dispatcher(int32_t, int32_t, intptr_t, void*, float)' at ../../dpf/distrho/src/DistrhoPluginVST.cpp:465:43:
../../dpf/distrho/src/DistrhoPluginVST.cpp:79:17: warning: 'char* strncpy(char*, const char*, size_t)' output may be truncated copying 23 bytes from a string of length 32 [-Wstringop-truncation]
     std::strncpy(dst, src, size-1);
     ~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~
```

Since the custom `DISTRHO::strncpy` function inserts a null char at `size-1` in any case, can't we just use `std::memcpy` instead of `std::strncpy` to get rid of the warning?

--- a/distrho/src/DistrhoPluginVST.cpp
+++ b/distrho/src/DistrhoPluginVST.cpp
@@ -76,7 +76,7 @@ static const writeMidiFunc writeMidiCallback = nullptr;
 
 void strncpy(char* const dst, const char* const src, const size_t size)
 {
-    std::strncpy(dst, src, size-1);
+    std::memcpy(dst, src, std::min(std::strlen(src), size-1));
     dst[size-1] = '\0';
 }
 

--- a/distrho/src/DistrhoPluginVST.cpp
+++ b/distrho/src/DistrhoPluginVST.cpp
@@ -76,18 +76,21 @@ static const writeMidiFunc writeMidiCallback = nullptr;
 
 void strncpy(char* const dst, const char* const src, const size_t size)
 {
+    DISTRHO_SAFE_ASSERT_RETURN(size > 0,);
     std::memcpy(dst, src, std::min(std::strlen(src), size-1));
     dst[size-1] = '\0';
 }
 
 void snprintf_param(char* const dst, const float value, const size_t size)
 {
+    DISTRHO_SAFE_ASSERT_RETURN(size > 0,);
     std::snprintf(dst, size-1, "%f", value);
     dst[size-1] = '\0';
 }
 
 void snprintf_iparam(char* const dst, const int32_t value, const size_t size)
 {
+    DISTRHO_SAFE_ASSERT_RETURN(size > 0,);
     std::snprintf(dst, size-1, "%d", value);
     dst[size-1] = '\0';
 }

